src/HIVCohortFilter/HIVCohortCases.jsx

```diff
@@ -0,0 +1,1205 @@
+import CohortPTCSvg from '../img/cohort-PTC.svg';
+import CohortECSvg from '../img/cohort-EC.svg';
+import CohortLTNPSvg from '../img/cohort-LTNP.svg';
+import Spinner from '../components/Spinner';
+
+class HIVCohortFilterCase extends React.Component {
+  // Base class for the 3 NDH cohort filter apps. Meant to facilitate code reuse
+  constructor(props) {
+    super(props);
+    this.updateFilters = this.updateFilters.bind(this);
+    this.checkReadyToCalculate = this.checkReadyToCalculate.bind(this);
+    this.downloadControl = this.downloadControl.bind(this);
+    this.therapyValuesOfInterest = ['HAART'];
+  }
+
+  async getFollowUpsWithHIV() {
+    const keyName = 'subject_id';
+    const keyCountMap = await this.getBucketByKey(keyName);
+    let batchCounts = 0;
+    const promiseList = [];
+    let keyRange = [];
+    Object.keys(keyCountMap).forEach((keyId) => {
+      const count = keyCountMap[keyId];
+      if (batchCounts + count > 5000) {
+        // query this batch for follow ups
+        promiseList.push(this.getFollowupsBuckets(keyName, keyRange).then(res => res));
+
+        // reset batch
+        batchCounts = count;
+        keyRange = [];
+      }
+      batchCounts += count;
+      keyRange.push(keyId);
+    });
+    promiseList.push(this.getFollowupsBuckets(keyName, keyRange).then(res => res));
+
+    let mergedFollowUps = [];
+    const resultList = await Promise.all(promiseList);
+    resultList.forEach((res) => {
+      mergedFollowUps = mergedFollowUps.concat(res);
+    });
+    return mergedFollowUps;
+  }
+
+  getFollowupsBuckets(key, keyRange) {
+    const query = `
+    {
+      follow_up {
+        hits(filters: { op: "and",
+          content: [
+            { op: "=",
+              content: { field: "hiv_status", value: "positive" }
+            },
+            {
+              op: "in",
+              content: { field: "${key}", value: ["${keyRange.join('","')}"] }
+            }
+          ]
+        }, first:10000) {
+          total
+          edges {
+            node {
+              subject_id
+              visit_number
+              thrpyv
+              visit_date
+              fposdate
+              frstdthd
+              leu3n
+              submitter_id
+              viral_load
+            }
+          }
+
+        }
+      }
+    }`;
+    return this.performQuery(query).then((res) => {
+      if (!res || !res.data) {
+        throw new Error('Error while querying subjects with HIV');
+      }
+      return res.data.follow_up.hits.edges.map(edge => edge.node);
+    });
+  }
+
+  performQuery(queryString) { // eslint-disable-line
+    return fetchWithCreds({
+      path: `${arrangerGraphqlPath}`,
+      body: JSON.stringify({
+        query: queryString,
+      }),
+      method: 'POST',
+    })
+      .then(
+        ({ status, data }) => data, // eslint-disable-line no-unused-vars
+      );
+  }
+
+  makeSubjectToVisitMap(followUps) { // eslint-disable-line class-methods-use-this
```

Comment:
what are you disabling with eslint here?

Reply:
class-methods-use-this

Reply:
this is like a "static" method kinda. a helper function. but i didnt mark it as static because it seems like that means something else in react

Reply:
right, but why all the eslint disabling?

Reply:
removed this eslint-disable ✓ 
but i left a few in the codebase. there are some that i don't know how to fix. for example: 
```
return fetchWithCreds({
      path: `${arrangerGraphqlPath}`,
      body: JSON.stringify({
        query: queryString,
      }),
      method: 'POST',
    })
      .then(
        ({ status, data }) => data, // eslint-disable-line no-unused-vars
      );
```

eslint talks about "status" being unused. but i can't remove that variable :O that's how callbacks go right?